tests/microos/toolbox.pm

```diff
@@ -0,0 +1,85 @@
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Run simple toolbox tests
+# Maintainer: Jose Lausuch <jalausuch@suse.com>
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use version_utils 'is_sle_micro';
+
+
+sub run {
+    select_console 'root-console';
+    my $user = 'test_user';
+
+    # Create user and don't ask password for sudo commands
+    assert_script_run "useradd -m $user ";
+    assert_script_run "echo \"$user ALL=(ALL) NOPASSWD:ALL\" >> /etc/sudoers";
+
+    # Display help
+    record_soft_failure 'https://github.com/kubic-project/microos-toolbox/issues/22' if (script_run('toolbox -h') != 0);
+
+    record_info 'Test',                    "Run toolbox without flags";
+    assert_script_run 'toolbox id',        timeout => 180;
+    validate_script_output 'podman ps -a', sub { m/toolbox-root/ };
+    assert_script_run 'podman rm toolbox-root';
+
+    record_info 'Test',                         "Run toolbox with a given tag";
+    assert_script_run 'toolbox -t test_tag id', timeout => 180;
+    validate_script_output 'podman ps -a',      sub { m/toolbox-root-test_tag/ };
+    assert_script_run 'podman rm toolbox-root-test_tag';
+
+    record_info 'Test', "Run toolbox with a given name";
+    if (script_run('toolbox -c test_name id') == 0) {
+        validate_script_output 'podman ps -a', sub { m/test_name/ };
+        assert_script_run 'podman rm test_name';
+    } else {
+        record_soft_failure 'https://github.com/kubic-project/microos-toolbox/issues/19';
+    }
+
+    record_info 'Test', "Rootless toolbox as $user";
+    my $prefix = "runuser -l $user -c";
+    my $uid    = script_output "$prefix 'id -u'";
+    validate_script_output "$prefix 'toolbox -u id'", sub { m/uid=${uid}\(${user}\)/ }, timeout => 180;
+    die "$user shouldn't have access to /etc/passwd!" if (script_run("$prefix 'toolbox -u touch /etc/passwd'") == 0);
+    assert_script_run "$prefix \"podman rm toolbox-$user-user\"";
+
+    record_info 'Test',                               "Rootfull toolbox as $user";
+    validate_script_output "$prefix 'toolbox -r id'", sub { m/uid=0\(root\)/ };
+    die "Root should have access to /etc/passwd!" if (script_run("$prefix 'toolbox -r touch /etc/passwd'") != 0);
+    assert_script_run "podman rm toolbox-$user";
+
+
+    record_info 'Test', 'Pulling toolbox image from different registry';
+    # Switch default registries for openSUSE MicroOS and SLE Micro
+    my $registry = is_sle_micro ? 'registry.opensuse.org'   : 'registry.suse.com';
+    my $image    = is_sle_micro ? 'opensuse/toolbox:latest' : 'suse/sle-micro/5.0/toolbox:latest';
+    my $expected = is_sle_micro ? 'opensuse'                : 'sles';
+    assert_script_run 'echo -e "REGISTRY=' . $registry . '\nIMAGE=' . $image . '" > ~/.toolboxrc';
+    validate_script_output 'toolbox cat /etc/os-release', sub { m/${expected}/ }, timeout => 180;
```

Comment:
why not `check_os_release($expected, 'PRETTY_NAME', 'toolbox');` would it work?

Reply:
tried and it doesn't work:

cause it does `toolbox grep -e '^PRETTY_NAME\b' /etc/os-release`
and it complains `toolbox: invalid option -- 'e'`.
So, we would need to tweak the function to achieve issuing `toolbox -- grep -e '^PRETTY_NAME\b' /etc/os-release`.